synthesizer/process/src/finalize.rs

```diff
@@ -183,6 +192,101 @@ fn finalize_fee_transition<N: Network, P: FinalizeStorage<N>>(
     }
 }
 
+/// Finalizes the constructor.
+fn finalize_constructor<N: Network, P: FinalizeStorage<N>>(
+    state: FinalizeGlobalState,
+    store: &FinalizeStore<N, P>,
+    stack: &Stack<N>,
+    transition_id: N::TransitionID,
+) -> Result<Vec<FinalizeOperation<N>>> {
+    // Retrieve the program ID.
+    let program_id = stack.program_id();
+    #[cfg(debug_assertions)]
+    println!("Finalizing constructor for {}...", stack.program_id());
+
+    // Initialize a list for finalize operations.
+    let mut finalize_operations = Vec::new();
+
+    // Initialize a nonce for the constructor registers.
+    // Currently, this nonce is set to zero for every constructor.
+    let nonce = 0;
+
+    // Get the constructor logic. If the program does not have a constructor, return early.
+    let Some(constructor) = stack.program().constructor() else {
+        return Ok(finalize_operations);
+    };
+
+    // Get the constructor types.
+    let constructor_types = stack.get_constructor_types()?.clone();
+
+    // Initialize the finalize registers.
+    let mut registers = FinalizeRegisters::new(state, transition_id, *program_id.name(), constructor_types, nonce);
```

Comment:
Auditors: The parameters used to initialize the FinalizeRegisters here should be reviewed for correctness and soundness.

Reply:
@d0cd make sure to go over all unresolved comments and copy them over in the new PR you're making.

Reply:
I think we're all resolved in this PR no?